x-pack/legacy/plugins/infra/public/pages/logs/analysis/page_setup_content.tsx

```diff
@@ -4,13 +4,212 @@
+);
+const endTimeDefaultDescription = i18n.translate(
+  'xpack.infra.analysisSetup.endTimeDefaultDescription',
+  {
+    defaultMessage: 'indefinitely',
+  }
+);
+const clearStartTimeLabel = i18n.translate('xpack.infra.analysisSetup.clearStartTimeLabel', {
+  defaultMessage: 'Clear start time',
+});
+const clearEndTimeLabel = i18n.translate('xpack.infra.analysisSetup.clearEndTimeLabel', {
+  defaultMessage: 'Clear end time',
+});
+
+function selectedDateToParam(selectedDate: Moment | null) {
+  if (selectedDate) {
+    return selectedDate.unix();
+  }
+  return undefined;
+}
+
+export const AnalysisSetupContent: React.FunctionComponent<AnalysisSetupContentProps> = ({
+  setupMlModule,
+  isSettingUp,
+}) => {
   useTrackPageview({ app: 'infra_logs', path: 'analysis_setup' });
   useTrackPageview({ app: 'infra_logs', path: 'analysis_setup', delay: 15000 });
 
-  return <div>Setup</div>;
+  const [startTime, setStartTime] = useState<Moment | null>(null);
+  const [endTime, setEndTime] = useState<Moment | null>(null);
+
+  const now = useMemo(() => moment(), []);
+  const selectedEndTimeIsToday = !endTime || endTime.isSame(now, 'day');
+
+  const onClickCreateJob = () =>
+    setupMlModule(selectedDateToParam(startTime), selectedDateToParam(endTime));
+
+  return (
+    <AnalysisSetupPage>
+      <EuiPageBody>
+        <AnalysisPageContent
+          verticalPosition="center"
+          horizontalPosition="center"
+          data-test-subj="analysisSetupContent"
+        >
+          <EuiPageContentHeader>
+            <EuiPageContentHeaderSection>
+              <EuiTitle size="m">
+                <h3>
+                  <FormattedMessage
+                    id="xpack.infra.analysisSetup.analysisSetupTitle"
+                    defaultMessage="Enable Machine Learning analysis"
+                  />
+                </h3>
+              </EuiTitle>
+            </EuiPageContentHeaderSection>
+          </EuiPageContentHeader>
+          <EuiPageContentBody>
+            <EuiText size="s">
+              <FormattedMessage
+                id="xpack.infra.analysisSetup.analysisSetupDescription"
+                defaultMessage="Use Machine Learning to automatically categorize log messages, and detect anomalous event counts and rare log messages."
+              />
+            </EuiText>
+            <EuiSpacer size="l" />
+            <EuiDescribedFormGroup
+              idAria="timeRange"
+              title={
+                <FormattedMessage
+                  id="xpack.infra.analysisSetup.timeRangeTitle"
+                  defaultMessage="Choose a time range"
+                />
+              }
+              description={
+                <FormattedMessage
+                  id="xpack.infra.analysisSetup.timeRangeDescription"
+                  defaultMessage="By default, Machine Learning will analyze log messages starting now and continue indefinitely. You can specify an earlier date to begin, a specific point in time to end, or both."
```

Comment:
By default data will be collected from the start of the index, so I'd change this to:

```suggestion
                  defaultMessage="By default, Machine Learning will analyze log messages from the start of your log indices and continue indefinitely. You can specify an earlier date to begin, to end, or both."
```

Reply:
Just spotted this in my inbox. The present tense may be better than the future " . . . Machine Learning **analyzes** log messages from the start of your log indices and **continues** indefinitely."? Otherwise it reads like this behaviour will happen at some point in the future when we get round to implementing it. 

And, being a bit pedantic, if the analysis starts from the start of the logs, is it meaningful to specify an even earlier date to start the analysis from? Or in this case, is it just a **different** date? Presumably this date could be later than the start of the logs, but earlier than now?

Reply:
> if the analysis starts from the start of the logs, is it meaningful to specify an even earlier date to start the analysis from? Or in this case, is it just a **different** date? Presumably this date could be later than the start of the logs, but earlier than now?

Yeah that's correct. Should change "earlier", good catch.